iina/Extensions.swift

```diff
@@ -163,6 +163,16 @@ extension Array {
   }
 }
 
+extension Dictionary {
+  mutating func safeAppend<T: Equatable>(_ value: T, forKey key: Key) where Value == Array<T> {
+    if self[key] == nil {
+      self[key] = Array<T>()
```

Comment:
With more thought, I think we should just provide an overloaded subscript function that automatically initializes any key's value when queried:
```swift
extension Dictionary where Value: ExpressibleByArrayLiteral {
	subscript(autogenerating key: Key) -> Value {
		mutating get {
			if let value = self[key] {
				return value
			} else {
				self[key] = []
				return self[autogenerating: key]
			}
		}
	}
}
```
Unfortunately this is blocked by [SR-3547](https://bugs.swift.org/browse/SR-3547), so the above solution is the best we can do for now.